ERCS/erc-7674.md

```diff
@@ -0,0 +1,56 @@
+---
+eip: 7674
+title: Transient Approval Extension for ERC-20
+description: An interface for ERC-20 approvals via transient storage
+author: Xenia Shape (@byshape), Mikhail Melnik (@ZumZoom), Hadrien Croubois (@Amxx)
+discussions-to: https://ethereum-magicians.org/t/add-erc-7674-transient-approval-extension-for-erc-20/19521
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-02
+requires: 20, 1153
+---
+
+## Abstract
+
+This specification defines the minimum interface required to transiently approve `ERC-20` tokens for spending within a single transaction.
+
+## Motivation
+
+`EIP-1153` allows to use a cheaper way to transiently store allowances.
+
+## Specification
+
+The key words "MUST", "SHOULD", "MAY" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Compliant contracts MUST implement 1 new function in addition to `ERC-20`:
+```solidity
+function transientApprove(address spender, uint256 value) public returns (bool success)
+```
+Call to `transientApprove(spender, value)` allows `spender` to withdraw within the same transaction from `msg.sender` multiple times, up to the `value` amount.
+
+Compliant contracts MUST use the transient storage `EIP-1153` to keep the temporary allowance. For each `owner` and `spender`, the slot MUST be uniquely selected to avoid slot collision. Each slot index SHOULD be derived from the base slot index for transient allowances, `owner` and `spender` addresses. Slot MAY be derived as `keccak256(spender . keccak256(owner . p))` where `.` is concatenation and `p` is `keccak256` from the string uniquely defining transient allowances in the namespace of the implementing contract.
+
+Compliant contracts MUST add a transient allowance check to the `transferFrom` function. The permanent allowance can only be spent after the temporary allowance has been exhausted.
+
+Compliant contracts MUST add a transient allowance to the permanent one when returning the allowed amount to spend in the `allowance` function.
```

Comment:
```suggestion
Compliant contracts MUST add a transient allowance to the permanent one when returning the allowed amount to spend in the `allowance` function. In case the sum of the transient and permanent allowance overflow, `type(uint256).max` SHOULD be returned.
```